brainiak/reprsimil/brsa.py

```diff
@@ -344,34 +617,170 @@ def fit(self, X, design, nuisance=None, scan_onsets=None, coords=None,
+            (recommended) when fitting the model, data should be z-scored
+            as well when calling transform()
+        y : not used (as it is unsupervised learning)
+        scan_onsets : numpy array, shape=[number of runs]. Default: None.
+            A list of indices corresponding to the onsets of
+            scans in the data X. If not provided, data will be assumed
+            to be acquired in a continuous scan.
+        Returns
+        -------
+        ts : numpy arrays, shape = [time_points, condition]
+            The estimated response to the task conditions which have the
+            response amplitudes estimated during the fit step.
+        ts0: numpy array, shape = [time_points, n_nureg]
+            The estimated time course spread across the brain, with the
+            loading weights estimated during the fit step.
+        """
+
+        assert X.ndim == 2 and X.shape[1] == self.beta_.shape[1], \
+            'The shape of X is not consistent with the shape of data '\
+            'used in the fitting step. They should have the same number '\
+            'of voxels'
+        assert scan_onsets is None or (scan_onsets.ndim == 1 and
+                                       0 in scan_onsets), \
+            'scan_onsets should either be None or an array of indices '\
+            'If it is given, it should include at least 0'
+
+        if scan_onsets is None:
+            scan_onsets = np.array([0], dtype=int)
+        else:
+            scan_onsets = np.int32(scan_onsets)
+        ts, ts0, log_p = self._transform(
+            Y=X, scan_onsets=scan_onsets, beta=self.beta_,
+            beta0=self.beta0_, rho_e=self.rho_, sigma_e=self.sigma_,
+            rho_X=self._rho_design_, sigma2_X=self._sigma2_design_,
+            rho_X0=self._rho_X0_, sigma2_X0=self._sigma2_X0_)
+        return ts, ts0
+
+    def score(self, X, design, scan_onsets=None):
+        """ Use the model and parameters estimated by fit function
+            from some data of a participant to evaluate the log
+            likelihood of some new data of the same participant.
+            Design matrix of the same set of experimental
+            conditions in the testing data should be provided, with each
+            column corresponding to the same condition as that column
+            in the design matrix of the training data.
+            Unknown nuisance time series will be marginalized, assuming
+            they follow the same spatial pattern as in the training
+            data. The hypothetical response captured by the design matrix
+            will be subtracted from data before the marginalization
+            when evaluating the log likelihood. For null model,
+            nothing will be subtracted before marginalization.
+            If you z-scored your data during fit step, you should
+            z-score them for score function as well. If you did not
+            z-score in fitting, you should not z-score here either.
```

Comment:
I decided to still leave it for the user to do z-scoring of the whole data or not before fitting. I personall feel that if training and testing data have slightly different variance in each voxel, z-scoring them separately might cause different scaling among different voxels, then transform() and score() might be impacted, especially transform() which relies on the assumption that the patterns of activation/nuisance activity stays constant across runs.